message/validation.go

```diff
@@ -16,35 +17,39 @@ type Validator func(*watermillmessage.Message) (*ValidationError, error)
 
 // ValidationError represents a message validation error.
 type ValidationError struct {
-	Timestamp time.Time                 `json:"ts"`
-	Msg       *watermillmessage.Message `json:"msg"`
-	Errors    []string                  `json:"errors"`
+	Timestamp time.Time `json:"ts"`
+	Errors    []string  `json:"errors"`
 }
 
 func (v ValidationError) Error() string {
-	return fmt.Sprintf("message %q is invalid. Validation errors: %s", v.Msg.UUID, strings.Join(v.Errors, " | "))
+	return strings.Join(v.Errors, " | ")
 }
 
 // NewValidationError creates a new ValidationError.
-func NewValidationError(msg *watermillmessage.Message, ts time.Time, errors ...string) *ValidationError {
-	return &ValidationError{Msg: msg, Timestamp: ts, Errors: errors}
+func NewValidationError(ts time.Time, errors ...string) *ValidationError {
+	return &ValidationError{Timestamp: ts, Errors: errors}
 }
 
-// ValidationErrorNotifier notifies whenever a ValidationError happens.
-type ValidationErrorNotifier func(validationError *ValidationError) error
-
-// ValidationErrorToChanNotifier notifies to a given chan when a ValidationError happens.
-func ValidationErrorToChanNotifier(errChan chan *ValidationError) ValidationErrorNotifier {
```

Comment:
All of this has been removed in favor of using publisher through watermill.io routes.